squad/api/rest.py

```diff
@@ -762,15 +762,24 @@ class EnvironmentViewSet(ModelViewSet):
     ordering_fields = ('id', 'slug', 'name')
 
 
+class HyperlinkedMetricIdentityField(serializers.HyperlinkedIdentityField):
+    def get_url(self, *args):
+        testrun = args[0]
+        if testrun.metrics.count() == 0:
+            return None
+        else:
+            return super().get_url(*args)
+
+
 class TestRunSerializer(serializers.HyperlinkedModelSerializer):
 
     id = serializers.IntegerField(read_only=True)
     tests_file = serializers.HyperlinkedIdentityField(view_name='testrun-tests-file')
-    metrics_file = serializers.HyperlinkedIdentityField(view_name='testrun-metrics-file')
+    metrics_file = HyperlinkedMetricIdentityField(view_name='testrun-metrics-file')
```

Comment:
any reason to not also do the same for `tests_file`? it seems inconsitent that one will always have the link while the other may not.

Reply:
@terceiro yes was just talking this thought with @chaws today, the reason i held off, was thinking that no sensible testrun would carry no tests unless it is not completed. Any requestor for results for downstream analysis would most likely query completed ones ...**i think**..... We aren't sure! But it makes sense from uniformity and compliance PoV to have both. What do you think?

Reply:
@amrohassaan there can be a testrun with only metrics and no tests.